staging/src/k8s.io/apiserver/pkg/server/filters/timeout.go

```diff
@@ -146,13 +146,13 @@ func (t *timeoutHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			return false
 		})
 		defer postTimeoutFn()
-		tw.timeout(err)
+		tw.timeout(r, err)
 	}
 }
 
 type timeoutWriter interface {
 	http.ResponseWriter
-	timeout(*apierrors.StatusError)
+	timeout(*http.Request, *apierrors.StatusError)
```

Comment:
we had a race in this handler, https://github.com/kubernetes/kubernetes/commit/44705c71401d327c6d596597adc55596973e89d0 , we need to check plumbing the request is ok or if we should use the clone `rCopy`

Reply:
or just plumb the exact bits we need (context and protomajor)